shared/common-adapters/native-image.native.tsx

```diff
@@ -34,51 +35,54 @@ class FastImageImpl extends React.Component<
     this._mounted = false
   }
 
-  _onLoadStart = () => {
+  private onLoadStart = () => {
     this._mounted && this.setState({loading: true})
-    this.props.onLoadStart && this.props.onLoadStart()
+    this.props.onLoadStart?.()
   }
-  _onLoadEnd = () => {
+  private onLoadEnd = () => {
     this._mounted && this.setState({loading: false})
-    this.props.onLoadEnd && this.props.onLoadEnd()
+    this.props.onLoadEnd?.()
   }
-  _onProgress = evt => {
+  private onProgress = (evt: OnProgressEvent) => {
     this._mounted && this.setState({progress: evt.nativeEvent?.loaded / evt.nativeEvent?.total || 0})
-    this.props.onProgress && this.props.onProgress(evt)
+    this.props.onProgress?.(evt as any)
   }
+
+  private getStyle = memoize((style, loading) => {
+    return Styles.collapseStyles([style, loading ? styles.loading : styles.loaded] as const)
+  }, isEqual)
+
+  private getSource = memoize(source =>
+    isArray(source)
+      ? {uri: source[0].uri} // TODO smarter choice?
+      : {uri: source.uri}
+  )
+
   render() {
     if (typeof this.props.source === 'number') {
       return null
     }
 
-    let source: ImageURISource
-    if (isArray(this.props.source)) {
-      source = this.props.source[0] // TODO smarter choice?
-    } else {
-      source = this.props.source
-    }
+    const source = this.getSource(this.props.source)
 
-    if (!source || !source.uri) {
+    if (!source.uri) {
       return null
     }
+
+    // TODO maybe use reanimated2
```

Comment:
what's the story here?

Reply:
we could use reanimated more instead of the built in animated

Reply:
(not worth doing now)